vcd/remove_leftovers_test.go

```diff
@@ -42,6 +42,8 @@ var doNotDelete = entityList{
 	{Type: "vcd_vapp", Name: "Test_EmptyVmVapp1", Comment: "created by test, but to be preserved"},
 	{Type: "vcd_vapp", Name: "Test_EmptyVmVapp2", Comment: "created by test, but to be preserved"},
 	{Type: "vcd_vapp", Name: "Test_EmptyVmVapp3", Comment: "created by test, but to be preserved"},
+	{Type: "vcd_solution_add_on", Name: "vmware.solution-addon-landing-zone-1.2.0-22957452", Comment: "Built-in Solution Add-On"},
+	{Type: "vcd_solution_add_on", Name: "vmware.autoscale-1.4.0-23130968", Comment: "Built-in Solution Add-On"},
```

Comment:
I was told they didn't change for a long time AFAIK, but in fact what I am doing in next PRs is that I introduced a mechanism to match by regexp so that one doesn't have to hardcode exact versions as this isn't practical with Solution Add-On instances (as well as there might be some other cases)

Reply:
Backported the mechanism here (where it belongs) and ran tests as well as cleanup. Looks to be working.